### PR TITLE
Consistent formatting of flags with args

### PR DESCRIPTION
**Pull request by richsalz**
For documentation of all commands with "-flag arg" format them
consistently: "B\<-flag> I\<arg>"
Update find-doc-nits to complain if badly formatted strings are found.

### CODE REVIEW

**Comment by richsalz** [comment]
Fixes #9987

File: doc/man1/CA.pl.pod [hunk 1]
```diff
@@ -106,7 +106,7 @@ Extra params are passed on to B<openssl x509> and B<openssl ca> commands.
 
 Generate a CRL. Executes B<openssl ca> command.
 
-=item B<-revoke certfile [reason]>
+=item B<-revoke> I<certfile [reason]>
```
[the remark this answers is not in the document]
Comment (richsalz) [reply doc/man1/CA.pl.pod]: fixed.

File: doc/man1/fipsinstall.pod [hunk 2]
```diff
@@ -62,21 +62,21 @@ Standard input is used if the filename is '-'.
...
 
-=item B<-mac_name name>
+=item B<-mac_name> I<name>
 
 Specifies the name of a supported MAC algorithm which will be used.
 To see the list of supported MAC's use the command I<list -mac-algorithms>.
 The default is "HMAC".
 
-=item B<-macopt nm:v>
+=item B<-macopt> I<nm:v>
```
[the remark this answers is not in the document]
Comment (richsalz) [reply doc/man1/fipsinstall.pod]: fixed.

File: doc/man1/openssl-asn1parse.pod [hunk 3]
```diff
@@ -8,21 +8,21 @@ openssl-asn1parse - ASN.1 parsing tool
 
 B<openssl> B<asn1parse>
 [B<-help>]
-[B<-inform PEM|DER>]
-[B<-in filename>]
-[B<-out filename>]
+[B<-inform> I<DER|PEM>]
```
[the remark this answers is not in the document]
Comment (richsalz) [reply doc/man1/openssl-asn1parse.pod]: fixed, they are now ```I<DER>|I<PEM>``` throughout.

File: doc/man1/openssl-asn1parse.pod [hunk 4]
```diff
@@ -143,7 +143,7 @@ Some knowledge of the ASN.1 structure is needed to interpret the output.
 
 In this example the BIT STRING at offset 229 is the certificate public key.
 The contents octets of this will contain the public key information. This can
-be examined using the option B<-strparse 229> to yield:
+be examined using the option B<-strparse> I<229> to yield:
```
[the remark this answers is not in the document]
Comment (richsalz) [reply doc/man1/openssl-asn1parse.pod]: makes sense, fixed.

File: doc/man1/openssl-ca.pod [hunk 5]
```diff
@@ -9,53 +9,53 @@ openssl-ca - sample minimal CA application
...
-[B<-extfile section>]
-[B<-engine id>]
-[B<-subj arg>]
+[B<-extensions> I<section>]
+[B<-extfile> I<section>]
+[B<-engine> I<id>]
+[B<-subj> I<arg>]
 [B<-utf8>]
-[B<-sigopt nm:v>]
+[B<-sigopt> I<nm:v>]
```
[the remark this answers is not in the document]
Comment (richsalz) [reply doc/man1/openssl-ca.pod]: fixed throughout.

File: doc/man1/openssl-ca.pod [hunk 6]
```diff
@@ -9,53 +9,53 @@ openssl-ca - sample minimal CA application
...
 [B<-create_serial>]
 [B<-rand_serial>]
 [B<-multivalue-rdn>]
-[B<-rand file...>]
-[B<-writerand file>]
+[B<-rand> I<file...>]
```
[the remark this answers is not in the document]
Comment (richsalz) [reply doc/man1/openssl-ca.pod]: removing nonsense is outside the scope of this PR :)  no, okay, I fixed it :)

**Comment by richsalz** [comment]
I dislike the ```B<fooI<TYPE>bar>``` construct being used in some other pages, but I made a change to accomodate it.  I also rewrite a sentence to avoid using ```B<foo<I<*>>```

**Comment by richsalz** [comment]
All other issues fixed.

File: doc/man1/openssl-cms.pod [hunk 7]
```diff
@@ -50,25 +52,25 @@ B<openssl> B<cms>
...
+[B<-verify_depth> I<num>]
+[B<-verify_email> I<email>]
+[B<-verify_hostname> I<hostname>]
+[B<-verify_ip> I<ip>]
+[B<-verify_name> I<name>]
 [B<-x509_strict>]
-[B<-md digest>]
-[B<-I<cipher>>]
+[B<-md> I<digest>]
+[B<->I<cipher>]
```
Comment (levitte) [inline doc/man1/openssl-cms.pod]: Hmmm, **-*cipher*** vs <b>-</b>*cipher*. I prefer the former, because that shows that the cipher is an actual option rather than an argument for `-`.
Semantic subtlety.

File: doc/man1/openssl-dgst.pod [hunk 8]
```diff
@@ -7,25 +7,25 @@ openssl-dgst - perform digest operations
 =head1 SYNOPSIS
 
 B<openssl dgst>
-[B<-I<digest>>]
+[B<->I<digest>]
```
Comment (levitte) [inline doc/man1/openssl-dgst.pod]: Same comment as for **-*cipher***

File: doc/man1/openssl-dgst.pod [hunk 9]
```diff
@@ -7,25 +7,25 @@ openssl-dgst - perform digest operations
...
+[B<-signature> I<filename>]
+[B<-sigopt> I<nm>:I<v>]
+[B<-hmac> I<key>]
 [B<-fips-fingerprint>]
-[B<-rand file...>]
-[B<-engine id>]
+[B<-rand> I<file...>]
+[B<-engine> I<id>]
 [B<-engine_impl>]
 [B<file...>]
```
Comment (levitte) [inline doc/man1/openssl-dgst.pod]: Should be italics, not bold

File: doc/man1/openssl-dgst.pod [hunk 10]
```diff
@@ -52,7 +52,7 @@ command.
 
 Print out a usage message.
 
-=item B<-I<digest>>
+=item B<-> I<digest>
```
Comment (levitte) [inline doc/man1/openssl-dgst.pod]: And this shows why this change is a bad idea. That space is *plain wrong*

File: doc/man1/openssl-dsaparam.pod [hunk 11]
```diff
@@ -8,17 +8,17 @@ openssl-dsaparam - DSA parameter manipulation and generation
...
 [B<-C>]
-[B<-rand file...>]
-[B<-writerand file>]
+[B<-rand> I<file...>]
+[B<-writerand> I<file>]
 [B<-genkey>]
-[B<-engine id>]
+[B<-engine> I<id>]
 [B<-verbose>]
 [B<numbits>]
```
Comment (levitte) [inline doc/man1/openssl-dsaparam.pod]: Should be italics, not bold

File: doc/man1/openssl-enc.pod [hunk 12]
```diff
@@ -6,37 +6,38 @@ openssl-enc - symmetric cipher routines
 
 =head1 SYNOPSIS
 
-B<openssl enc -I<cipher>>
+B<openssl>
+[B<->I<cipher>]
```
Comment (levitte) [inline doc/man1/openssl-enc.pod]: Again

File: doc/man1/openssl-genpkey.pod [hunk 13]
```diff
@@ -8,14 +8,14 @@ openssl-genpkey - generate a private key
...
-[B<-pass arg>]
-[B<-I<cipher>>]
-[B<-engine id>]
-[B<-paramfile file>]
-[B<-algorithm alg>]
-[B<-pkeyopt opt:value>]
+[B<-out> I<filename>]
+[B<-outform> I<DER>|I<PEM>]
+[B<-pass> I<arg>]
+[B<->I<cipher>]
```
Comment (levitte) [inline doc/man1/openssl-genpkey.pod]: Again

File: doc/man1/openssl-genpkey.pod [hunk 14]
```diff
@@ -8,14 +8,14 @@ openssl-genpkey - generate a private key
...
+[B<-engine> I<id>]
+[B<-paramfile> I<file>]
+[B<-algorithm> I<alg>]
+[B<-pkeyopt> I<opt:value>]
```
Comment (levitte) [inline doc/man1/openssl-genpkey.pod]: `I<opt>:I<value>`

File: doc/man1/CA.pl.pod [hunk 15]
```diff
@@ -106,7 +106,7 @@ Extra params are passed on to B<openssl x509> and B<openssl ca> commands.
 
 Generate a CRL. Executes B<openssl ca> command.
 
-=item B<-revoke certfile [reason]>
+=item B<-revoke> I<certfile> I<[reason]>
```
[the remark this answers is not in the document]
Comment (richsalz) [reply doc/man1/CA.pl.pod]: done.

File: doc/man1/openssl-ca.pod [hunk 16]
```diff
@@ -9,53 +9,53 @@ openssl-ca - sample minimal CA application
...
+[B<-sigopt> I<nm>:I<v>]
 [B<-create_serial>]
 [B<-rand_serial>]
 [B<-multivalue-rdn>]
-[B<-rand file...>]
-[B<-writerand file>]
+[B<-rand> I<file>]
+[B<-writerand> I<file>]
 [B<-sm2-id string>]
 [B<-sm2-hex-id hex-string>]
```
[the remark this answers is not in the document]
Comment (richsalz) [reply doc/man1/openssl-ca.pod]: merged after this PR; fixed.

File: doc/man1/openssl-cms.pod [hunk 17]
```diff
@@ -78,28 +80,28 @@ B<openssl> B<cms>
...
-[B<-inkey file>]
-[B<-keyopt name:parameter>]
-[B<-passin arg>]
-[B<-rand file...>]
-[B<-writerand file>]
+[B<-secretkey> I<key>]
+[B<-secretkeyid> I<id>]
+[B<-econtent_type> I<type>]
+[B<-inkey> I<file>]
+[B<-keyopt> I<name:parameter>]
```
[the remark this answers is not in the document]
Comment (richsalz) [reply doc/man1/openssl-cms.pod]: There are a zillion of those.  For a separate PR if someone wants to do them.

File: doc/man1/openssl-cms.pod [hunk 18]
```diff
@@ -78,28 +80,28 @@ B<openssl> B<cms>
...
+[B<-passin> I<arg>]
+[B<-rand> I<file...>]
+[B<-writerand> I<file>]
 [B<cert.pem...>]
```
[the remark this answers is not in the document]
Comment (richsalz) [reply doc/man1/openssl-cms.pod]: No, the cert is an option recipient cert (zero or more)

File: doc/man1/openssl-cms.pod [hunk 19]
```diff
@@ -78,28 +80,28 @@ B<openssl> B<cms>
...
-[B<-to addr>]
-[B<-from addr>]
-[B<-subject subj>]
+[B<-to> I<addr>]
+[B<-from> I<addr>]
+[B<-subject> I<subj>]
 [cert.pem]...
```
[the remark this answers is not in the document]
Comment (richsalz) [reply doc/man1/openssl-cms.pod]: space before ellipsis is not something we do.  Separate PR.

File: doc/man1/openssl-dgst.pod (hunk 10 above)
Comment (richsalz) [reply doc/man1/openssl-dgst.pod]: space removed.

File: doc/man1/openssl-dsaparam.pod (hunk 11 above)
Comment (richsalz) [reply doc/man1/openssl-dsaparam.pod]: what should be italics?

File: doc/man1/openssl-enc.pod (hunk 12 above)
Comment (richsalz) [reply doc/man1/openssl-enc.pod]: again, disagree.  someone can fix that in a separate PR.

File: doc/man1/openssl-genpkey.pod (hunk 13 above)
Comment (richsalz) [reply doc/man1/openssl-genpkey.pod]: again.

File: doc/man1/openssl-genpkey.pod (hunk 14 above)
Comment (richsalz) [reply doc/man1/openssl-genpkey.pod]: separate PR if anyone's interested.

File: doc/man1/openssl-cms.pod (hunk 18 above)
[the remark this answers is not in the document]
Comment (levitte) [reply doc/man1/openssl-cms.pod]: So, er, what about the one just a few lines down?

File: doc/man1/openssl-dsaparam.pod (hunk 11 above)
Comment (levitte) [reply doc/man1/openssl-dsaparam.pod]: `[B<numbits>]` should be `[I<numbits>]`

File: doc/man1/openssl-enc.pod (hunk 12 above)
Comment (levitte) [reply doc/man1/openssl-enc.pod]: So we're going to have you "fixing", just for someone else to "unfix"?

File: util/find-doc-nits [hunk 20]
```diff
@@ -255,6 +255,20 @@ sub check {
...
+        err($id, "Bad flag formatting inside B<>")
+            if $contents =~ /B<-[A-Za-z_ ]+ /;
+        while ( $contents =~ /([BI])<([^>]*)>/g ) {
+            my $B = $1;
+            my $T = $2;
+            next if $T =~ /E</;  # Assume it's E<lt>
+            err($id, "Bad content inside $B<$T>")
+                if $T =~ /[<|]/;
+        }
+    }
```
Comment (levitte) [inline util/find-doc-nits]: You is checking the *whole* content, right?  That's gonna hurt...  see, DESCRIPTION is free flow text, and both bold and italics are allowed to be used for *emphasis*.  Sure, there are conventions on how names that are the subject of the document and how variables / arguments should be typeset, but this doesn't mean the emphasis is suddenly disallowed.

SYNOPSIS is not as free flow, as it's meant as a summary.  We can be strict there.

File: doc/man7/bio.pod [hunk 21]
```diff
@@ -47,10 +47,10 @@ in a memory leak.
 Calling BIO_free_all() on a single BIO has the same effect as calling
 BIO_free() on it other than the discarded return value.
 
-Normally the I<type> argument is supplied by a function which returns a
+Often the I<type> argument is supplied to a function which returns a
```
Comment (levitte) [inline doc/man7/bio.pod]: Uhmmmmm, no?  The "*type*" argument is the argument to `BIO_new()`, and for example `BIO_f_base64()` supplies such a "*type*", so the change of `by` to `to` is incorrect.

**Comment by levitte** [comment]
As for the `BIO_ADDR.pod` changes, where you remove the emphasis.  Typographically speaking, I'm actually fine with it, the emphasis isn't really needed.  However, you say this in another PR:

> The changes in ```BIO_ADDR.pod``` might be controversial, please take a look.  The other option is to remove the embedded ```L<>``` tag, or allow them, or something else.

This indicates that decisions are made, not so much because of style, and more on the grounds of technical limitations in `util/find-doc-nits`.  That's not a good reason to limit expressive freedom, which *should* be allowed as long as it doesn't blatantly break with conventions we want to follow.

File: util/find-doc-nits (hunk 20 above)
Comment (richsalz) [reply util/find-doc-nits]: There are no issues with the script right now.  If someone feels a crying need to do ```B<thisI<really>word>``` and doc-nits complains, we can fix the script.

Optimize the common case: it's much more common to be an error.

**Comment by richsalz** [comment]
> fix the script first.

I'm kinda done with this PR.

File: doc/man7/bio.pod (hunk 21 above)
Comment (richsalz) [reply doc/man7/bio.pod]: I still think that sentence needs to be rewritten, but I guess that will wait for another change. I undid my change here.

File: doc/man1/openssl-enc.pod (hunk 12 above)
Comment (richsalz) [reply doc/man1/openssl-enc.pod]: > So we're going to have you "fixing", just for someone else to "unfix"?

It won't be the first time and it most definitely won't be the last.

The changes are consistent and I consider that more important.  Others may disagree, that's okay.

File: doc/man1/openssl-dsaparam.pod (hunk 11 above)
Comment (richsalz) [reply doc/man1/openssl-dsaparam.pod]: thanks for the explanation, fixed.

File: doc/man1/openssl-cms.pod (hunk 18 above)
[the remark this answers is not in the document]
Comment (richsalz) [reply doc/man1/openssl-cms.pod]: oh!  oops.  fixed, thanks.

**Comment by richsalz** [comment]
Feedback addressed, or I disagreed, or I'm just plain done. :)

**Comment by richsalz** [comment]
This PR has 1400 changes over 54 files.  There are currently seven commits, but I am sure that I have squashed/rebased multiple times.

I think this is a huge improvement over what we currently have. It's not perfect, more could be done, but isn't that always the case?

**Comment by richsalz** [comment]
I also removed the "Fixes" text from the commits since the scope of this has changed and it no longer fixes everything that people (well at least me and @levitte:) think should be done to address that issue.

File: util/find-doc-nits (hunk 20 above)
Comment (levitte) [reply util/find-doc-nits]: Actually, I would say that there *is* an issue with this script right now.  It had you remove an emphasis in `doc/man3/BIO_ADDR.pod`, which is *actually perfectly fine* in that description.  I'm not terribly attached to that emphasis, but it does serve as an example where this script *does too much*.

I cannot see myself approving this on that ground.

Comment (richsalz) [reply util/find-doc-nits]: I think the emphasis was (being polite) incorrect. Again, you think this does too much and I think it catches things that are more likely to be wrong.

Sorry you're not going to approve (after you've done all that review work), but that happens.  I think optimizing the common case -- the use is wrong -- is more important.  Shrug.

Someone else can take this PR as the basis for their own PR, of course.

File: doc/man1/openssl-asn1parse.pod [hunk 22]
```diff
@@ -8,21 +8,21 @@ openssl-asn1parse - ASN.1 parsing tool
 
 B<openssl> B<asn1parse>
 [B<-help>]
-[B<-inform PEM|DER>]
-[B<-in filename>]
-[B<-out filename>]
+[B<-inform> I<DER>|I<PEM>]
```
Comment (t8m) [inline doc/man1/openssl-asn1parse.pod]: Why should DER/PEM be in italics?

File: doc/man1/openssl-asn1parse.pod [hunk 23]
```diff
@@ -37,16 +37,16 @@ structures. It can also be used to extract data from ASN.1 formatted data.
 
 Print out a usage message.
 
-=item B<-inform> B<DER|PEM>
+=item B<-inform> B<DER>|I<PEM>
```
Comment (t8m) [inline doc/man1/openssl-asn1parse.pod]: Again here shouldn't it be B<PEM>, or it should be at least consistent.

File: doc/man1/openssl-rehash.pod [hunk 24]
```diff
@@ -11,12 +11,12 @@ openssl-c_rehash - Create symbolic links to files named by the hash values
 
 B<openssl>
 B<rehash>
-B<[-h]>
-B<[-help]>
-B<[-old]>
-B<[-n]>
-B<[-v]>
-[ I<directory>...]
+B<-h>
```
Comment (t8m) [inline doc/man1/openssl-rehash.pod]: Why are the  brackets removed here?

File: doc/man1/openssl-s_client.pod [hunk 25]
```diff
@@ -112,32 +112,32 @@ B<openssl> B<s_client>
...
 [B<-ct>]
 [B<-noct>]
 [B<-ctlogfile>]
-[B<-keylogfile file>]
-[B<-early_data file>]
+[B<-keylogfile> I<file>]
+[B<-early_data> I<file>]
 [B<-enable_pha>]
-[B<target>]
+[I<host>:I<post>]
```
Comment (t8m) [inline doc/man1/openssl-s_client.pod]: s/post/port/

Comment (t8m) [reply doc/man1/openssl-s_client.pod]: And should be actually `[I<host>[:I<port>]]`

File: doc/man1/openssl-asn1parse.pod (hunk 22 above)
Comment (richsalz) [reply doc/man1/openssl-asn1parse.pod]: Because it is a literal argument.  ping @levitte for backup :)

File: doc/man1/openssl-asn1parse.pod (hunk 23 above)
Comment (richsalz) [reply doc/man1/openssl-asn1parse.pod]: made to be ```I<>```  And fixed the markup in the text description in the lines that follow.

File: doc/man1/openssl-rehash.pod (hunk 24 above)
Comment (richsalz) [reply doc/man1/openssl-rehash.pod]: Because I goofed, it should be ```[B<-goo>]```.  good catch.  fixed.

File: doc/man1/openssl-s_client.pod (hunk 25 above)
Comment (richsalz) [reply doc/man1/openssl-s_client.pod]: there's a bunch of places which would need to be changed for that.  Just on this manpage, the -bind, -proxy, -connect.  As well as the ocsp and s_time pages.  I prefer to leave that for another PR (and presumably someone else).

**Comment by richsalz** [comment]
Commit pushed to address @t8m's feedback.

Again, please squash this when merging.

File: doc/man1/openssl-asn1parse.pod (hunk 22 above)
Comment (levitte) [reply doc/man1/openssl-asn1parse.pod]: Oh wait, damn, those *should* be bold. The reason is that they are names to be replaced some values, but literal values. Damn.

Comment (richsalz) [reply doc/man1/openssl-asn1parse.pod]: That is not the answer I was hoping for, Richard. :)  Ah well.  Fixup commit pushed.  It only fixes PEM DER ENGINE SMIME etc.  If there are others... I don't want to know.

File: doc/man1/openssl-s_client.pod (hunk 25 above)
Comment (t8m) [reply doc/man1/openssl-s_client.pod]: OK

**Comment by levitte** [comment]
Sure

**Comment by levitte** [comment]
Merged.

e8769719c9bbe53d7af088111b7625671660d4db Consistent formatting of flags with args

**Comment by richsalz** [comment]
Thanks everyone, particularly @levitte, for all the detail work on reviews.
